tests/gen/README.md

```diff
@@ -0,0 +1,156 @@
+# Automatic Documents Generation 
+
+`AutoGenerate` and `AutoGenerateFromSDL` are a highly versatile function designed for dynamic document generation, perfect for testing and simulation purposes. 
+
+`AutoGenerateFromSDL` creates documents based on a specified schema, allowing for extensive customization of data generation. 
+
+The function generates documents adhering to a defined schema and it's configuration.
+It interprets the types and relationships within the schema to create realistic, interconnected data structures.
+
+`AutoGenerate` creates documents based on the provider collections' definitions (`[]client.CollectionDefinition`)
+
+### Demand Calculation:
+
+The functions calculate the 'demand' or the number of documents to generate based on the configuration provided.
+For related types within the schema, it intelligently adjusts the number of generated documents to maintain consistency in relationships (one-to-one, one-to-many, etc.).
```

Comment:
todo: Similar to other points, but I think that `within the schema, ` is incorrect here, and should instead read something like `within the collection set, `